Content.Client/PDA/Ringer/RingtoneMenu.xaml.cs

```diff
@@ -0,0 +1,105 @@
+                }
+                else {
+                    previousNoteInputs[0] = NoteText1.Text;
+                }
+
+                NoteText1.CursorPosition = NoteText1.Text.Length; //Resets caret position to the end of the typed input
+            };
+
+            //RingerNoteTwoInput
+            var NoteText2 = this.RingerNoteTwoInput;
+
+            NoteText2.OnTextChanged += _ => //Prevents unauthorized characters from being entered into the LineEdit
+            {
+                NoteText2.Text = NoteText2.Text.ToUpper();
+
+                if(!isNote(NoteText2.Text)) {
+                    NoteText2.Text = previousNoteInputs[1];
+                }
+                else {
+                    previousNoteInputs[1] = NoteText2.Text;
+                }
+
+                NoteText2.CursorPosition = NoteText2.Text.Length; //Resets caret position to the end of the typed input
+            };
+
+            //RingerNoteThreeInput
+            var NoteText3 = this.RingerNoteThreeInput;
+
+            NoteText3.OnTextChanged += _ => //Prevents unauthorized characters from being entered into the LineEdit
+            {
+                NoteText3.Text = NoteText3.Text.ToUpper();
+
+                if(!isNote(NoteText3.Text)) {
+                    NoteText3.Text = previousNoteInputs[2];
+                }
+                else {
+                    previousNoteInputs[2] = NoteText3.Text;
+                }
+
+                NoteText3.CursorPosition = NoteText3.Text.Length; //Resets caret position to the end of the typed input
+            };
+
+            //RingerNoteFourInput
+            var NoteText4 = this.RingerNoteFourInput;
+
+            NoteText4.OnTextChanged += _ => //Prevents unauthorized characters from being entered into the LineEdit
+            {
+                NoteText4.Text = NoteText4.Text.ToUpper();
+
+                if(!isNote(NoteText4.Text)) {
+                    NoteText4.Text = previousNoteInputs[3];
+                }
+                else {
+                    previousNoteInputs[3] = NoteText4.Text;
+                }
+
+                NoteText4.CursorPosition = NoteText4.Text.Length; //Resets caret position to the end of the typed input
+            };
+        }
+
+        protected override DragMode GetDragModeFor(Vector2 relativeMousePos)
+        {
+            //Prevents the ringtone window from being resized
+            return DragMode.Move;
+        }
+
+        private bool isNote(string input) //Determines whether or not the characters inputed are authorized
+        {
+            bool isNoteOut = false;
+            string[] noteList = new string[13]{"A","B","C","D","E","F","G","A#","C#","D#","F#","G#",""};
+            for(int arrCount = 0; arrCount < 13; arrCount++) {
+                if(input.Equals(noteList[arrCount])) {
+                    isNoteOut = true;
+                }
+            }
+            return isNoteOut;
+        }
```

Comment:
Like I said elsewhere when you just make it an enum you can just check if the integer is < 12 and not have this expensive and uggo check. You also don't do the validation on the server and a malicious client can just bypass this.